components/feature/pwa/src/main/java/mozilla/components/feature/pwa/WebAppLauncherActivity.kt

```diff
@@ -22,34 +26,48 @@ import mozilla.components.support.ktx.kotlin.toUri
  */
 class WebAppLauncherActivity : AppCompatActivity() {
     private val logger = Logger("WebAppLauncherActivity")
+    private lateinit var storage: ManifestStorage
+    private val activityJob = Job()
```

Comment:
_Not a big issue for not, but may be important for future modifications._

I suggest using `SupervisorJob` here (to prevent cancelling all children if one of them failed).

Or event better (I think), something like:
```kotlin
class WebAppLauncherActivity : AppCompatActivity(), CoroutineScope by MainScope() {

    override fun onDestroy() {
        coroutineContext.cancel()
        super.onDestroy()
    }
}
```

Reply:
TIL, thanks!